Sources/SwiftProtobuf/ProtobufBinarySizeVisitor.swift

```diff
@@ -0,0 +1,119 @@
+        let tagSize = Varint.encodedSize(of: UInt32(
+            truncatingBitPattern: protoFieldNumber << 3 | M.protobufWireType()))
+        let messageSize = try value.serializedSize()
+        serializedSize += tagSize + Varint.encodedSize(of: UInt64(messageSize)) + messageSize
+    }
+
+    mutating func visitRepeatedMessageField<M: ProtobufMessage>(value: [M], protoFieldNumber: Int, protoFieldName: String, jsonFieldName: String, swiftFieldName: String) throws {
+        let tagSize = Varint.encodedSize(of: UInt32(
+            truncatingBitPattern: protoFieldNumber << 3 | M.protobufWireType()))
+        serializedSize += value.count * tagSize
+        for v in value {
+            let messageSize = try v.serializedSize()
+            serializedSize += Varint.encodedSize(of: UInt64(messageSize)) + messageSize
+        }
+    }
+
+    mutating func visitSingularGroupField<G: ProtobufGroup>(value: G, protoFieldNumber: Int, protoFieldName: String, jsonFieldName: String, swiftFieldName: String) throws {
+        // The wire format doesn't matter here because the encoded size of the integer won't change
+        // based on the low three bits.
+        let tagSize = Varint.encodedSize(of: UInt32(truncatingBitPattern: protoFieldNumber << 3))
+        serializedSize += 2 * tagSize
+        try withAbstractVisitor {(visitor: inout ProtobufVisitor) in
+            try value.traverse(visitor: &visitor)
+        }
+    }
+
+    mutating func visitRepeatedGroupField<G: ProtobufGroup>(value: [G], protoFieldNumber: Int, protoFieldName: String, jsonFieldName: String, swiftFieldName: String) throws {
+        let tagSize = Varint.encodedSize(of: UInt32(truncatingBitPattern: protoFieldNumber << 3))
+        serializedSize += 2 * value.count * tagSize
+        for v in value {
+            try withAbstractVisitor {(visitor: inout ProtobufVisitor) in
+                try v.traverse(visitor: &visitor)
+            }
+        }
+    }
+
+    mutating func visitMapField<KeyType: ProtobufMapKeyType, ValueType: ProtobufMapValueType>(fieldType: ProtobufMap<KeyType, ValueType>.Type, value: ProtobufMap<KeyType, ValueType>.BaseType, protoFieldNumber: Int, protoFieldName: String, jsonFieldName: String, swiftFieldName: String) throws where KeyType.BaseType: Hashable {
+        let tagSize = Varint.encodedSize(of: UInt32(truncatingBitPattern: protoFieldNumber << 3))
+        let keyTagSize = Varint.encodedSize(of: UInt32(truncatingBitPattern: 1 << 3))
+        let valueTagSize = Varint.encodedSize(of: UInt32(truncatingBitPattern: 2 << 3))
+        for (k,v) in value {
+            let entrySize = try keyTagSize + KeyType.encodedSize(of: k) + valueTagSize + ValueType.encodedSize(of: v)
+            serializedSize += entrySize + Varint.encodedSize(of: Int64(entrySize))
+        }
+        serializedSize += value.count * tagSize
+    }
+}
```

Comment:
This PR adds `serializedSize()` on messages [here](https://github.com/apple/swift-protobuf/pull/43/files#diff-fcfd4c47681aa25d602aab7dba779084R753), which does *not* include the length varint before it. There's also the `encodedSize()` method in the various binary type protocols that does include it, but that's not exposed to clients. Did you mean the first or the second one?